feat(LoneScale Node): Add LoneScale node and Trigger node

SanYann opened the pull request:
Github issue / Community forum post (link here to close automatically):

SanYann committed:
feat(LoneScale Nodes): Add trigger node and manage lists node

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-lonescale

Joffcom commented:
Hey @SanYann,

I was just starting to review this and I got part of the way through it and I was wondering if this was finished or still a work in progress?

Joffcom committed:
Improvements

SanYann commented:
> Hey @SanYann,
> 
> I was just starting to review this and I got part of the way through it and I was wondering if this was finished or still a work in progress?

Hello, It is finished and tested locally

Joffcom commented:
Hey @SanYann,

Perfect, I had to double check as there were some things that were a bit odd. It looks like the linter is still showing some errors and there is no support for `continueOnFail`. I have added in a credential test and moved all of the requests to use the same function.

When I was looking at the UI I noticed that adding a company and a contact uses the same fields, Is that expected? Looking at the API docs there might not be a first name field for company. Do you know if there is a way to get hold of test credentials or would signing up for a trial be the best bet?

SanYann commented:
> Hey @SanYann,
> 
> Perfect, I had to double check as there were some things that were a bit odd. It looks like the linter is still showing some errors and there is no support for `continueOnFail`. I have added in a credential test and moved all of the requests to use the same function.
> 
> When I was looking at the UI I noticed that adding a company and a contact uses the same fields, Is that expected? Looking at the API docs there might not be a first name field for company. Do you know if there is a way to get hold of test credentials or would signing up for a trial be the best bet?

Hello,

Thanks a lot for your improvements.
Company and People uses the same fields for now, you can for exemple create a people attached to a company.
To test credentials, I can create a test account with default password on your email address, you will just have to create a new one on your first connection

Joffcom commented:
Hey @SanYann,

That could work, There is still a bit more tidying up to do and support for features like continue on fail and paired items but I think another couple of sessions and it should be in a good place.

SanYann commented:
> continueOnFail

Do you have documentation I could follow or example to implements  continue on fail and paired items?

SanYann commented:
>

Joffcom commented:
Hey @SanYann,

PairedItems can be found here: https://docs.n8n.io/integrations/creating-nodes/build/reference/paired-items/

ContinueOnFail I can't see anything for but most nodes will have it at the bottom of the item loop, It will look something like this...

```
const executionData = this.helpers.constructExecutionMetaData(
					this.helpers.returnJsonArray(responseData),
					{ itemData: { item: i } },
				);
				returnData.push(...executionData);
			} catch (error) {
				if (this.continueOnFail()) {
					returnData.push({ error: error.message, json: {} });
					continue;
				}
				throw error;
			}
		}
```

If you use the newer declarative style this sort of thing is taken care of for you as well and it normally means you can build out a node fairly quickly as you don't need to worry about some of the more standard things but both styles are valid.

SanYann committed:
Add ContinueOnFail and paired items

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-lonescale

Joffcom commented on packages/nodes-base/credentials/LoneScaleApi.credentials.ts:
```diff
@@ -0,0 +1,38 @@
+		type: 'generic',
+		properties: {
+			headers: {
+				Authorization: '=Bearer {{$credentials.apiKey}}',
```
This header key should be...
```
'X-API-KEY': '={{$credentials.apiKey}}',
```
Using the Authorization header with Bearer leads to an authentication failure

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				},
+				options: [
+					{
+						name: 'Add',
```
We prefer Create over Add for operation names

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+						name: 'Company',
+						value: 'COMPANY',
+						description: 'List of company',
+						action: 'Get company lists',
```
This is a create not a get

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+					{
+						name: 'Contact',
+						value: 'PEOPLE',
+						description: 'List of contact',
```
As above

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				required: true,
+			},
+			{
+				displayName: 'First Name',
```
This should be marked as required as it is needed to create a person on a list

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact first name',
+			},
+			{
+				displayName: 'Last Name',
```
This should be marked as required as it is needed to create a person on a list

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact last name',
+			},
+			{
+				displayName: 'Full Name',
```
This is optional, Should be moved under an Additional Items field

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact full name',
+			},
+			{
+				displayName: 'Linkedin Url',
```
This is optional for a person, Should be moved under an Additional Items field

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact Linkedin URL',
+			},
+			{
+				displayName: 'Company Name',
```
This is optional for a person, Should be moved under an Additional Items field. For a Company it should be marked as required.

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact current position',
+			},
+			{
+				displayName: 'Company Domain',
```
This is optional for both person and company, Should be moved under an Additional Items field

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				description: 'Contact company domain',
+			},
+			{
+				displayName: 'Contact Location',
```
This is optional for both person and company, Should be moved under an Additional Items field

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+				default: '',
+			},
+			{
+				displayName: 'Contact Email',
```
This is optional, Should be moved under an Additional Items field

Joffcom commented on packages/nodes-base/nodes/LoneScale/LoneScaleList.node.ts:
```diff
@@ -0,0 +1,426 @@
+							entity,
+						};
+
+						console.log({ body });
```
This probably isn't needed and can be removed

SanYann committed:
Add Additional fields

Joffcom committed:
slight wording update

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into node-lonescale

Joffcom committed:
Add codex files

janober commented:
Got released with [`n8n@0.230.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.230.1)
